ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/tree/nodes/pairs/TomlKeyValue.kt

```diff
@@ -145,6 +145,7 @@ public fun String.parseValue(lineNo: Int, config: TomlInputConfig): TomlValue =
         // ===== basic strings
         '\"' -> TomlBasicString(this, lineNo)
         else -> tryParseValue<NumberFormatException>(lineNo, ::TomlLong)  // ==== integer values
+            ?: tryParseValue<NumberFormatException>(lineNo, ::TomlUnsignedLong)  // ===== unsigned integer values
```

Comment:
It looks like we will miss sometimes here, and I do not know, how to fix this situation. 

For example, we receive: UnsignedLog with value "1". How would we parse it? 

To SignedLong, right? :(

Reply:
> For example, we receive: UnsignedLong with value "1". How would we parse it?

As TomlLong, but then we can convert it to ULong. But now I see what I missed:
- When we have UByte/UShort/UInt, and toml contains negative number (let's say -1), then we got an error: "**The unsigned integer literal, that you have provided is <18446744073709551615>, but the type for deserialization**"
- When we have ULong and "-1" in toml, then the result is **18446744073709551615**

So I think we just need to add a check on decoding level to prevent this. If we have negative number and Unsigned type - throw an exception